### PR TITLE
SUS-3225: script to revert double htmlescapechars on wordmark-text

### PR DESCRIPTION
**Pull request by harnash**
ping: @Wikia/core-platform-team @Wikia/sus 

TODO: I'll run script in dry-run mode to see what changes will be applied.

### CODE REVIEW

**Review by mszabo-wikia — APPROVED** [review]

File: maintenance/wikia/WikiFactoryVariables/unescapeWordmarkText.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+			$cnt++;
+			$old_value = $value;
+			$value = Sanitizer::decodeCharReferences( $value );
+			$value = htmlspecialchars_decode( $value, ENT_QUOTES );
+		} while ( $value !== $old_value || $cnt >= 10 );
+
+		return $value;
+	}
+
+	public function execute() {
```
Comment (WikiaTech) [inline maintenance/wikia/WikiFactoryVariables/unescapeWordmarkText.php:48]: MAJOR Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[project-id]: wikia:app

File: maintenance/wikia/WikiFactoryVariables/unescapeWordmarkText.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		global $wgCityId, $wgMedusaHostPrefix;
```
Comment (WikiaTech) [inline maintenance/wikia/WikiFactoryVariables/unescapeWordmarkText.php]: MAJOR Remove this unused "$wgMedusaHostPrefix" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

[project-id]: wikia:app

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[project-id]: wikia:app
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: maintenance/wikia/WikiFactoryVariables/unescapeWordmarkText.php [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+		}
+
+		if ( !$this->dryRun ) {
+			$globalStateWrapper = new Wikia\Util\GlobalStateWrapper( [
+				'wgUser' => User::newFromName( Wikia::BOT_USER, false )
+			] );
+
+			$globalStateWrapper->wrap( function () use ( $themeSettings, $settings ) {
+				$themeSettings->saveSettings( $settings );
+			} );
```
Comment (macbre) [inline maintenance/wikia/WikiFactoryVariables/unescapeWordmarkText.php:112]: I think we can simplify the code here and just call `$wgUser = User::newFromName( Wikia::BOT_USER, false )` at the top of `execute` method.

File: maintenance/wikia/WikiFactoryVariables/unescapeWordmarkText.php [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+	/**
+	 * Print the message if verbose is enabled
+	 * @param $msg
+	 */
+	protected function debug( $msg ) {
+		if ( $this->verbose ) {
+			$this->output( $msg );
+		}
+	}
```
Comment (macbre) [inline maintenance/wikia/WikiFactoryVariables/unescapeWordmarkText.php]: You can rely on `output` method and call the script with `--quiet` param to silence these messages.

**Review by macbre — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[project-id]: wikia:app
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: maintenance/wikia/WikiFactoryVariables/unescapeWordmarkText.php (hunk 4 above)
Comment (harnash) [reply maintenance/wikia/WikiFactoryVariables/unescapeWordmarkText.php]: This and above fixed in 7474c0366bb

**Review by macbre — APPROVED** [review]

**Pull request merged**